### PR TITLE
Deprecate unnecessary public variables in L2

### PR DESCRIPTION
**Pull request by m-chrzan**
### Description

As part of cleanup for L2, we want to revert implicit functions created for `public` variables that will be deprecated on CEL2. This is done by:

- Renaming the variable to `deprecated_x`
- Making it `private`
- Adding a public, view, `onlyL1` function that returns the variable (for full compatibility until the L2 switch is flipped)

TODO:
- [x] Decide what still needs deprecated in GasPriceMinimum.

### Other changes

Minor cleanup in tests.

### Tested

Unit tests that verify the variables can still be retrieved on L1 but revert on L2.

### Related issues

- Fixes https://github.com/celo-org/celo-blockchain-planning/issues/653#issue-2564961955

### CODE REVIEW

**Comment by martinvol** [comment]
Everything in GasPriceMinimum should be deprecated on L2

File: packages/protocol/contracts/common/FeeCurrencyWhitelist.sol [hunk 1]
```diff
@@ -72,10 +84,10 @@ contract FeeCurrencyWhitelist is
    * @param index The index of the token in the whitelist array.
    */
   function removeToken(address tokenAddress, uint256 index) public onlyOwner {
```
Comment (martinvol) [inline packages/protocol/contracts/common/FeeCurrencyWhitelist.sol]: onlyL1 (just in case)

Comment (m-chrzan) [reply packages/protocol/contracts/common/FeeCurrencyWhitelist.sol]: Same for `addToken`, I assume?

**Review by soloseng — APPROVED** [review]

**Comment by soloseng** [comment]
> Everything in GasPriceMinimum should be deprecated on L2

+1

**Pull request merged**